src/test/regress/sql/multi_citus_tools.sql

```diff
@@ -208,9 +204,66 @@ SELECT * FROM master_run_on_worker(ARRAY[:node_name]::text[], ARRAY[:node_port]:
+	WHERE shardid % 2 = 0 AND nodeport = :worker_1_port;
+SELECT * FROM run_command_on_placements('check_placements', 'select 1');
+DROP TABLE check_placements CASCADE;
+
+-- make sure run_on_all_colocated_placements correctly detects colocation
+CREATE TABLE check_colocated (key int);
+SELECT master_create_distributed_table('check_colocated', 'key', 'hash');
+SELECT master_create_worker_shards('check_colocated', 5, 2);
+CREATE TABLE second_table (key int);
+SELECT master_create_distributed_table('second_table', 'key', 'hash');
+SELECT master_create_worker_shards('second_table', 4, 2);
+SELECT * FROM run_command_on_colocated_placements('check_colocated', 'second_table',
+												  'select 1');
+-- even when the difference is in replication factor, an error is thrown
+SELECT master_drop_all_shards('second_table'::regclass, current_schema(), 'second_table');
+SELECT master_create_worker_shards('second_table', 5, 1);
+SELECT * FROM run_command_on_colocated_placements('check_colocated', 'second_table',
+												  'select 1');
+-- when everything matches, the command is run!
+SELECT master_drop_all_shards('second_table'::regclass, current_schema(), 'second_table');
+SELECT master_create_worker_shards('second_table', 5, 2);
+SELECT * FROM run_command_on_colocated_placements('check_colocated', 'second_table',
+												  'select 1');
+-- when a placement is invalid considers the tables to not be colocated
+UPDATE pg_dist_shard_placement SET shardstate = 3 WHERE shardid = (
+		SELECT shardid FROM pg_dist_shard
+		WHERE nodeport = :worker_1_port AND logicalrelid = 'second_table'::regclass
+		ORDER BY 1 ASC LIMIT 1
+);
+SELECT * FROM run_command_on_colocated_placements('check_colocated', 'second_table',
+												  'select 1');
+-- when matching placement is also invalid, considers the tables to be colocated
+UPDATE pg_dist_shard_placement SET shardstate = 3 WHERE shardid = (
+		SELECT shardid FROM pg_dist_shard
+		WHERE nodeport = :worker_1_port AND logicalrelid = 'check_colocated'::regclass
+		ORDER BY 1 ASC LIMIT 1
+);
+SELECT * FROM run_command_on_colocated_placements('check_colocated', 'second_table',
+												  'select 1');
+DROP TABLE check_colocated CASCADE;
+DROP TABLE second_table CASCADE;
 
--- drop the function after use
-DROP FUNCTION master_run_on_worker(worker_name text[], port integer[], command text[],
-						  parallel boolean, OUT node_name text, OUT node_port integer,
-						  OUT success boolean, OUT result text);
+-- runs on all shards
+CREATE TABLE check_shards (key int);
+SELECT master_create_distributed_table('check_shards', 'key', 'hash');
+SELECT master_create_worker_shards('check_shards', 5, 2);
+SELECT * FROM run_command_on_shards('check_shards', 'select 1');
+UPDATE pg_dist_shard_placement SET shardstate = 3 WHERE shardid % 2 = 0;
+SELECT * FROM run_command_on_shards('check_shards', 'select 1');
+DROP TABLE check_shards CASCADE;
 
+-- set SHOW_CONTEXT back to default
+\set SHOW_CONTEXT errors
```

Comment:
GitHub is showing no newline at the end of this file, which I think most editors tend to add.